lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -65,5 +65,39 @@ sub latest_build {
     return $rs->get_column('value')->first;
 }
 
+=head2 latest_jobs
+
+=over
+
+=item Return value: array of only the most recent jobs in the resultset
+
+=back
+
+De-duplicates the jobs in the result set. Jobs are considered 'duplicates'
+if they are for the same DISTRI, VERSION, BUILD, TEST, FLAVOR, ARCH and
+MACHINE. For each set of dupes, only the latest job found is included in
+the return array.
+
+=cut
+sub latest_jobs {
+    my $self = shift;
+    my @jobs = $self->search(undef, {order_by => ['me.id DESC']});
+    my @latest;
```

Comment:
Isn't the while (my $job = $jobquery->next) method more efficient than loading everything into an array right away?